instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -422,6 +422,14 @@ def _instrument(self, **kwargs):
                     request.
         """
         lambda_handler = os.environ.get(ORIG_HANDLER, os.environ.get(_HANDLER))
+        if not lambda_handler:
```

Comment:
The env var in question is assigned to "/bin/sh" when using a docker lambda. So this corrects the issue when using a non-lambda container, but not for all cases. While not as elegant, a more through solution might be to perform the split and check the array size before unpacking.

Reply:
@mlorenzana please raise an issue and feel free to open a PR for your use-case